aquamarine/src/plumber.rs

```diff
@@ -269,25 +307,120 @@ impl<RT: AquaRuntime, F: ParticleFunctionStatic> Plumber<RT, F> {
     ) -> Poll<Result<RemoteRoutingEffects, AquamarineApiError>> {
         self.waker = Some(cx.waker().clone());
 
-        self.vm_pool.poll(cx);
+        self.poll_pools(cx);
 
         if let Some(event) = self.events.pop_front() {
             return Poll::Ready(event);
         }
 
-        // Gather effects and put VMs back
         let mut remote_effects: Vec<RemoteRoutingEffects> = vec![];
         let mut local_effects: Vec<LocalRoutingEffects> = vec![];
-        let mut interpretation_stats = vec![];
+        // Gather effects and put VMs back
+        self.process_host_actors(cx, &mut remote_effects, &mut local_effects);
+        self.process_workers_actors(cx, &mut remote_effects, &mut local_effects);
+
+        self.cleanup(cx);
+
+        // Execute next messages
+        let host_call_stats = self.process_next_host_messages(cx);
+        let workers_call_stats = self.process_next_worker_messages(cx);
```

Comment:
process => poll, and there are no messages there, so I suggest

```suggestion
        let host_call_stats = self.poll_next_host_actors(cx);
        let workers_call_stats = self.poll_next_worker_actors(cx);
```